AppKit/CoreAnimation/CPAnimationContext.j

```diff
@@ -152,24 +152,32 @@ CPLog.debug(_cmd + "context stack =" + _CPAnimationContextStack);
 
     duration = [animation duration] || [self duration];
 
-    var needsFrameTimer = (aKeyPath == @"frame" || aKeyPath == @"frameSize") &&
-                          ([anObject hasCustomLayoutSubviews] || [anObject hasCustomDrawRect]) &&
-                          (objectId = [anObject UID]);
+    var needsPeriodicFrameUpdates = (((aKeyPath === @"frame" || aKeyPath === @"frameSize") &&
+                                      ([anObject hasCustomLayoutSubviews] || [anObject hasCustomDrawRect])) ||
+                                     [[anObject animator] wantsPeriodicFrameUpdates]) &&
+                                    (objectId = [anObject UID]);
 
     if (_completionHandlerAgent)
         _completionHandlerAgent.increment();
 
     var completionFunction = function()
     {
-        if (needsFrameTimer)
+        if (needsPeriodicFrameUpdates)
+        {
             [self stopFrameUpdaterWithIdentifier:objectId];
+            if ([anObject respondsToSelector:@selector(_setForceUpdates:)])
+                [anObject _setForceUpdates:YES];
+        }
 
         if (animationCompletion)
             animationCompletion();
 
-        if (needsFrameTimer || animationCompletion)
+        if (needsPeriodicFrameUpdates || animationCompletion)
             [[CPRunLoop currentRunLoop] performSelectors];
 
+        if ([anObject respondsToSelector:@selector(_setForceUpdates:)])
+            [anObject _setForceUpdates:NO];
```

Comment:
I would prefer _setForceUpdate to be inside animationCompletion. If not it should be balanced with the same condition as the previous call.